### PR TITLE
[BEAM-6699] Configure artifact server port in DockerizedJobContainer

### PR DESCRIPTION
**Pull request by mxm**
On Docker-On-Mac, we need to explicitly set a port because we have to configure
port forwardings when bringing up the job server container. The artifact server
did not configure a port and we tried to use 0 as a port forwarding.

R @ryan-williams 

Post-Commit Tests Status (on master branch)
------------------------------------------------------------------------------------------------

Lang | SDK | Apex | Dataflow | Flink | Gearpump | Samza | Spark
--- | --- | --- | --- | --- | --- | --- | ---
Go | [Build Status](https://builds.apache.org/job/beam_PostCommit_Go/lastCompletedBuild/) | --- | --- | --- | --- | --- | ---
Java | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Apex/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Dataflow/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Flink/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Java_PVR_Flink_Batch/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Java_PVR_Flink_Streaming/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Gearpump/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Samza/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Spark/lastCompletedBuild/)
Python | [Build Status](https://builds.apache.org/job/beam_PostCommit_Python_Verify/lastCompletedBuild/) | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_Py_VR_Dataflow/lastCompletedBuild/) </br> [Build Status](https://builds.apache.org/job/beam_PostCommit_Py_ValCont/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PreCommit_Python_PVR_Flink_Cron/lastCompletedBuild/) | --- | --- | ---

See [.test-infra/jenkins/README](../.test-infra/jenkins/README.md) for trigger phrase, status and link of all Jenkins jobs.

### CODE REVIEW

File: sdks/python/apache_beam/runners/portability/portable_runner.py [hunk 1]
```diff
@@ -239,6 +240,7 @@ def send_options_request(max_retries=5):
           num_retries += 1
           if num_retries > max_retries:
             raise e
+          time.sleep(1)
```
Comment (mxm) [inline sdks/python/apache_beam/runners/portability/portable_runner.py]: FYI Included this here to give a bit more time for the channel to come up. Without waiting, it retries immediately and the retry is effectively worthless.

**Comment by ryan-williams** [comment]
Thanks for the quick fix @mxm, this lgtm (though I don't have great context on these code-paths).

Any idea why this suddenly broke? Do you think it's due to recent Docker upgrades?

**Comment by mxm** [comment]
It broke when we started including the wordcount in the PreCommit and made changes to let it bind to port 0, meaning a free port. When the job server runs inside docker on MacOs, we have to know the port in advance to work around the absence of host networking.

File: sdks/python/apache_beam/runners/portability/job_server.py [hunk 2]
```diff
@@ -57,9 +57,10 @@ def start(self):
            # "sibling" containers for the SDK harness.
            "-v", ':'.join([docker_path, "/bin/docker"]),
            "-v", "/var/run/docker.sock:/var/run/docker.sock"]
-    self.job_port = DockerizedJobServer._pick_port(self.job_port)
-    # artifact_port 0 suggest to pick a dynamic port.
-    self.artifact_port = self.artifact_port if self.artifact_port else 0
+
+    self.job_port, self.artifact_port = \
+        DockerizedJobServer._pick_port(self.job_port, self.artifact_port)
```
Comment (mxm) [inline sdks/python/apache_beam/runners/portability/job_server.py]: FYI, this is the relevant change.

**Pull request merged**